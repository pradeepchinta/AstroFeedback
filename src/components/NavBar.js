
import React from "react";
import { Navbar, Nav, Form, FormControl, Button } from "react-bootstrap";

const NavBar = ({ handleSearch, showSearch, searchText, handleChange, searchEnabled }) => {
    return (
        <Navbar bg="light" expand="lg">
            <Navbar.Brand href="/">Astro Customer Feedbacks</Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
                <Nav className="mr-auto">
                    <Nav.Link href="/add-new">Add New Feedback</Nav.Link>
                </Nav>
                {showSearch && <Form inline>
                    <FormControl type="text" placeholder="Search Using Email" className="mr-sm-2" value={searchText} onChange={handleChange}/>
                    <Button id="SearchButton" variant="outline-success" onClick={handleSearch} disabled={searchEnabled} >Search</Button>
                </Form>}
            </Navbar.Collapse>
        </Navbar>
    )
};

export default NavBar;