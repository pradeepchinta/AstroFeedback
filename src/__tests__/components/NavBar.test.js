import React from "react";
import { mount } from "enzyme";
import { Navbar } from "react-bootstrap";

import NavBar from "../../components/NavBar";


describe("NewFeedBack Component", () => {
    let wrapper;
    const minProps = {
        showSearch: true,
        handleSearch: jest.fn(),
    };

    it("will render with min props", () => {
        wrapper = mount(<NavBar {...minProps} />);
        expect(wrapper).toMatchSnapshot();
    })
    it("will render NavBar Brand", () => {
        expect(wrapper.find("NavBar").length).toBe(1);
        expect(wrapper.contains(<Navbar.Brand href="/">Astro Customer Feedbacks</Navbar.Brand>)).toBe(true)
    })
    it("will render Search Button", () => {
        expect(wrapper.find("Button").length).toBe(1);
        expect(wrapper.find("Button").props().children).toBe("Search");
    })
})