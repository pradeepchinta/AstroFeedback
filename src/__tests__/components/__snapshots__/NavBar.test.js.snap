// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewFeedBack Component will render with min props 1`] = `
<NavBar
  handleSearch={[MockFunction]}
  showSearch={true}
>
  <Navbar
    bg="light"
    collapseOnSelect={false}
    expand="lg"
    variant="light"
  >
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <NavbarBrand
        href="/"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          Astro Customer Feedbacks
        </a>
      </NavbarBrand>
      <NavbarToggle
        aria-controls="basic-navbar-nav"
        label="Toggle navigation"
      >
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
      </NavbarToggle>
      <NavbarCollapse
        id="basic-navbar-nav"
      >
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          id="basic-navbar-nav"
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <Transition
            addEndListener={[Function]}
            appear={false}
            aria-expanded={null}
            enter={true}
            exit={true}
            id="basic-navbar-nav"
            in={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={300}
            unmountOnExit={false}
          >
            <div
              aria-expanded={null}
              className="navbar-collapse collapse"
              id="basic-navbar-nav"
            >
              <Nav
                className="mr-auto"
                fill={false}
                justify={false}
              >
                <ForwardRef
                  as="div"
                  className="mr-auto navbar-nav"
                  onSelect={[Function]}
                >
                  <div
                    className="mr-auto navbar-nav"
                    onKeyDown={[Function]}
                  >
                    <NavLink
                      as={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "displayName": "SafeAnchor",
                          "render": [Function],
                        }
                      }
                      disabled={false}
                      href="/add-new"
                    >
                      <ForwardRef
                        as={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "displayName": "SafeAnchor",
                            "render": [Function],
                          }
                        }
                        className="nav-link"
                        disabled={false}
                        href="/add-new"
                      >
                        <SafeAnchor
                          className="nav-link"
                          data-rb-event-key="/add-new"
                          disabled={false}
                          href="/add-new"
                          onClick={[Function]}
                        >
                          <a
                            className="nav-link"
                            data-rb-event-key="/add-new"
                            href="/add-new"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                          >
                            Add New Feedback
                          </a>
                        </SafeAnchor>
                      </ForwardRef>
                    </NavLink>
                  </div>
                </ForwardRef>
              </Nav>
              <Form
                inline={true}
              >
                <form
                  className="form-inline"
                >
                  <FormControl
                    className="mr-sm-2"
                    placeholder="Search Using Email"
                    type="text"
                  >
                    <input
                      className="mr-sm-2 form-control"
                      placeholder="Search Using Email"
                      type="text"
                    />
                  </FormControl>
                  <Button
                    active={false}
                    disabled={false}
                    id="SearchButton"
                    onClick={[MockFunction]}
                    type="button"
                    variant="outline-success"
                  >
                    <button
                      className="btn btn-outline-success"
                      disabled={false}
                      id="SearchButton"
                      onClick={[MockFunction]}
                      type="button"
                    >
                      Search
                    </button>
                  </Button>
                </form>
              </Form>
            </div>
          </Transition>
        </Collapse>
      </NavbarCollapse>
    </nav>
  </Navbar>
</NavBar>
`;
